depmapomics/mutations.py

```diff
@@ -99,6 +76,61 @@ def addAnnotation(maf, NCBI_Build="37", Strand="+"):
     return maf
 
 
+def makeMatrices(maf, homin=0.95, id_col=SAMPLEID):
+    """ generates genotyped hotspot, driver and damaging mutation matrices
+
+    Returns:
+        hotspot_mat (pd.DataFrame): genotyped hotspot mutation matrix. 0 == not damaging, 1 == heterozygous, 2 == homozygous
+        lof_mat (pd.DataFrame): genotyped damaging mutation matrix. 0 == not damaging, 1 == heterozygous, 2 == homozygous
+        driver_mat (pd.DataFrame): genotyped driver mutation matrix. 0 == not driver, 1 == heterozygous, 2 == homozygous
+    """
+    print("generating genotyped driver and damaging mutation matrix")
+    gene_names = list(maf.hugo_symbol.unique())
+    hotspot_mat = pd.DataFrame(columns=gene_names)
+    lof_mat = pd.DataFrame(columns=gene_names)
+    driver_mat = pd.DataFrame(columns=gene_names)
+    sample_ids = list(maf[id_col].unique())
+    le = len(sample_ids)
+    for j in range(le):
```

Comment:
could be nice to decrease code use by have a *Arg that is a set of lambda functions then you just do subsetmaf[i] = maf[arg[i](maf)] for all i